src/python_testing/TC_SU_2_2.py

```diff
@@ -0,0 +1,1255 @@
+#     quiet: false
+# === END CI TEST ARGUMENTS ===
+
+import asyncio
+import logging
+import queue
+import time
+
+from mobly import asserts
+from TC_SUTestBase import SoftwareUpdateBaseTest
+
+import matter.clusters as Clusters
+from matter import ChipDeviceCtrl
+from matter.interaction_model import Status
+from matter.testing.event_attribute_reporting import AttributeMatcher, AttributeSubscriptionHandler, EventSubscriptionHandler
+from matter.testing.matter_testing import TestStep, async_test_body, default_matter_test_main
+
+# Create a logger
+logger = logging.getLogger(__name__)
+
+
+class TC_SU_2_2(SoftwareUpdateBaseTest):
+
+    async def add_single_ota_provider(self, controller, requestor_node_id: int, provider_node_id: int):
+        """
+        Adds a single OTA provider to the Requestor's DefaultOTAProviders attribute
+        only if no provider is currently registered. If a provider already exists,
+        the function does nothing.
+
+        Args:
+            controller: The device controller.
+            requestor_node_id (int): Node ID of the Requestor device.
+            provider_node_id (int): Node ID of the OTA Provider to add.
+
+        Returns:
+            None
+        """
+        # Read existing DefaultOTAProviders on the Requestor
+        current_providers = await self.read_single_attribute_check_success(
+            dev_ctrl=controller,
+            cluster=Clusters.OtaSoftwareUpdateRequestor,
+            attribute=Clusters.OtaSoftwareUpdateRequestor.Attributes.DefaultOTAProviders
+        )
+        logger.info(f'Prerequisite #4.0 - Current DefaultOTAProviders on Requestor: {current_providers}')
+
+        # If there is already a provider, skip adding
+        if current_providers:
+            logger.info(f'Skipping add: Requestor already has a provider registered ({current_providers})')
+            return
+
+        # Create a ProviderLocation for the new provider
+        provider_location = Clusters.OtaSoftwareUpdateRequestor.Structs.ProviderLocation(
+            providerNodeID=provider_node_id,
+            endpoint=0,
+            fabricIndex=controller.fabricId
+        )
+        logger.info(f'Prerequisite #4.0 - ProviderLocation to add: {provider_location}')
+
+        # Combine with existing providers (preserving previous ones)
+        updated_providers = current_providers + [provider_location]
+
+        # Write the updated DefaultOTAProviders list back to the Requestor
+        attr = Clusters.OtaSoftwareUpdateRequestor.Attributes.DefaultOTAProviders(value=updated_providers)
+        resp = await controller.WriteAttribute(
+            attributes=[(0, attr)],
+            nodeId=requestor_node_id
+        )
+        logger.info(f'Prerequisite #4.0 - Write DefaultOTAProviders response: {resp}')
+        asserts.assert_equal(resp[0].Status, Status.Success, "Failed to write DefaultOTAProviders attribute")
+
+    async def clear_ota_providers(self, controller, requestor_node_id: int):
```

Comment:
this also should be in the base. It would also make sense to add a call to this function in the teardown_test function in the base.